packages/neuron-wallet/src/models/chain/block-header.ts

```diff
@@ -0,0 +1,78 @@
+import TypeChecker from "utils/type-checker"
+
+export interface BlockHeaderInterface {
```

Comment:
I see no reason to have a `XxxInterface` layer.

My suggestion:

* Remove all these `Interfaces`.
* Expose class properties (read & write) by default.
* Comment or mark any properties that should have a setter to warn direct assignment.